starter_code/src/movies.js

```diff
@@ -1,19 +1,99 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(ary) {
+  let newAry = ary.map(function(mov) {
+    return Number(mov.rate);
+  });
+
+  let sumrates = newAry.reduce(function(Sumrates, varRate) {
+    return Sumrates + varRate;
+  }, 0);
+  return sumrates / newAry.length;
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(ary) {
+  if (!ary.length) return 0;
+  let cnt = 0;
+  let dramaSumRate = ary.reduce(function(Acc, value) {
+    if (value.genre.indexOf("Drama") !== -1) {
+      cnt++;
+      return Acc + Number(value.rate);
+    } else return Acc;
+  }, 0);
+
+  if (dramaSumRate === 0) return 0;
+  return Number((dramaSumRate / cnt).toFixed(2));
+}
 
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
 
+function orderByDuration(ary) {
+  if (!ary.length) return 0;
+  let sortMovies = ary.slice().sort(function(a, b) {
+    if (a.duration <= b.duration) return -1;
```

Comment:
This doesn't sort correctly by title as it doesn't respect the alphabetical order with different casing